app/src/main/java/com/kickstarter/viewmodels/ActivityFeedViewModel.java

```diff
@@ -61,180 +64,238 @@
+    private final Observable<SurveyResponse> goToSurvey;
+    private final BehaviorSubject<Boolean> isFetchingActivities= BehaviorSubject.create();
+    private final BehaviorSubject<Boolean> loggedInEmptyStateIsVisible = BehaviorSubject.create();
+    private final BehaviorSubject<Boolean> loggedOutEmptyStateIsVisible = BehaviorSubject.create();
+    private final BehaviorSubject<List<SurveyResponse>> surveys = BehaviorSubject.create();
 
-    public final Inputs inputs = this;
-    public final Outputs outputs = this;
+    public final ActivityFeedViewModel.Inputs inputs = this;
+    public final ActivityFeedViewModel.Outputs outputs = this;
 
     @Override public void emptyActivityFeedDiscoverProjectsClicked(final @NonNull EmptyActivityFeedViewHolder viewHolder) {
-      discoverProjectsClick.onNext(null);
+      this.discoverProjectsClick.onNext(null);
     }
 
     @Override public void emptyActivityFeedLoginClicked(final @NonNull EmptyActivityFeedViewHolder viewHolder) {
-      loginClick.onNext(null);
+      this.loginClick.onNext(null);
     }
 
     @Override public void friendBackingClicked(final @NonNull FriendBackingViewHolder viewHolder, final @NonNull Activity activity) {
-      friendBackingClick.onNext(activity);
+      this.friendBackingClick.onNext(activity);
     }
 
     @Override public void nextPage() {
-      nextPage.onNext(null);
+      this.nextPage.onNext(null);
     }
 
     @Override public void projectStateChangedClicked(final @NonNull ProjectStateChangedViewHolder viewHolder,
       final @NonNull Activity activity) {
-      projectStateChangedClick.onNext(activity);
+      this.projectStateChangedClick.onNext(activity);
     }
 
     @Override public void projectStateChangedPositiveClicked(final @NonNull ProjectStateChangedPositiveViewHolder viewHolder,
       final @NonNull Activity activity) {
-      projectStateChangedPositiveClick.onNext(activity);
+      this.projectStateChangedPositiveClick.onNext(activity);
     }
 
     @Override public void projectUpdateClicked(final @NonNull ProjectUpdateViewHolder viewHolder,
       final @NonNull Activity activity) {
-      projectUpdateClick.onNext(activity);
+      this.projectUpdateClick.onNext(activity);
     }
 
     @Override public void projectUpdateProjectClicked(final @NonNull ProjectUpdateViewHolder viewHolder,
       final @NonNull Activity activity) {
-      projectUpdateProjectClick.onNext(activity);
+      this.projectUpdateProjectClick.onNext(activity);
+    }
+
+    @Override public void surveyClicked(final @NonNull UnansweredSurveyViewHolder viewHolder, final @NonNull SurveyResponse surveyResponse) {
+      this.surveyClick.onNext(surveyResponse);
     }
 
     @Override public void refresh() {
-      refresh.onNext(null);
+      this.refresh.onNext(null);
+    }
+
+    @Override public void resume() {
+      this.resume.onNext(ActivityEvent.RESUME);
+    }
+
+    @Override @NonNull public Observable<List<Activity>> activities() {
```

Comment:
can you please audit these output functions for annotation order? as a reminder we like to do `public @NonNull` so that the annotation is right by the type it's annotating!